p2p/src/peer.rs

```diff
@@ -14,7 +14,7 @@
 
 use std::fs::File;
 use std::net::{SocketAddr, TcpStream};
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
```

Comment:
We should use the `Mutex` from parking_lot (as we do for `RwLock`) then we can avoid the unwrap() calls etc.

Reply:
👍